src/metabase/lib/remove_replace.cljc

```diff
@@ -377,15 +384,16 @@
     stage-number :- :int
     target-clause
     new-clause]
-   (cond
-     (and (map? target-clause) (= (:lib/type target-clause) :mbql/join))
-     (replace-join query stage-number target-clause new-clause)
+   (let [query (backfill-effective-type query)]
```

Comment:
Why is this necessary here? How do you get a query (built through `query-method`) without effective-type now?

Reply:
Oh, it is not necessary anymore. Fixing `query-method :mbql/query` did the trick. Thanks.